pkg/scheduler/in_memory_build_queue.go

```diff
@@ -46,129 +48,192 @@ import (
 			Help:      "Number of times that tasks were retried before completing.",
 			Buckets:   prometheus.LinearBuckets(0, 1, 11),
 		},
-		[]string{"instance_name_prefix", "platform", "size_class", "result", "grpc_code"})
-	inMemoryBuildQueueTasksCompletedDurationSeconds = prometheus.NewHistogramVec(
+		[]string{"result", "grpc_code"},
+	}
+
+	inMemoryBuildQueueTasksCompletedDurationSecondsTemplate = histogramTemplate{
 		prometheus.HistogramOpts{
 			Namespace: "buildbarn",
 			Subsystem: "builder",
 			Name:      "in_memory_build_queue_tasks_completed_duration_seconds",
 			Help:      "Time in seconds that tasks were completed before being removed.",
 			Buckets:   util.DecimalExponentialBuckets(-3, 6, 2),
 		},
-		[]string{"instance_name_prefix", "platform", "size_class"})
+		[]string{},
+	}
 
-	inMemoryBuildQueueWorkersCreatedTotal = prometheus.NewCounterVec(
+	inMemoryBuildQueueWorkersCreatedTotalTemplate = counterTemplate{
 		prometheus.CounterOpts{
 			Namespace: "buildbarn",
 			Subsystem: "builder",
 			Name:      "in_memory_build_queue_workers_created_total",
 			Help:      "Number of workers created by Synchronize().",
 		},
-		[]string{"instance_name_prefix", "platform", "size_class"})
-	inMemoryBuildQueueWorkersTerminatingTotal = prometheus.NewCounterVec(
+		[]string{},
+	}
+
+	inMemoryBuildQueueWorkersTerminatingTotalTemplate = counterTemplate{
 		prometheus.CounterOpts{
 			Namespace: "buildbarn",
 			Subsystem: "builder",
 			Name:      "in_memory_build_queue_workers_terminating_total",
 			Help:      "Number of workers that have entered the terminating state.",
 		},
-		[]string{"instance_name_prefix", "platform", "size_class"})
-	inMemoryBuildQueueWorkersRemovedTotal = prometheus.NewCounterVec(
+		[]string{},
+	}
+
+	inMemoryBuildQueueWorkersRemovedTotalTemplate = counterTemplate{
 		prometheus.CounterOpts{
 			Namespace: "buildbarn",
 			Subsystem: "builder",
 			Name:      "in_memory_build_queue_workers_removed_total",
 			Help:      "Number of workers removed due to expiration.",
 		},
-		[]string{"instance_name_prefix", "platform", "size_class", "state"})
+		[]string{"state"},
+	}
 
-	inMemoryBuildQueueWorkerInvocationStickinessRetained = prometheus.NewHistogramVec(
+	inMemoryBuildQueueWorkerInvocationStickinessRetainedTemplate = histogramTemplate{
 		prometheus.HistogramOpts{
 			Namespace: "buildbarn",
 			Subsystem: "builder",
 			Name:      "in_memory_build_queue_worker_invocation_stickiness_retained",
 			Help:      "How many levels of worker invocation stickiness were respected, as configured through worker_invocation_stickiness_limits.",
 			Buckets:   prometheus.LinearBuckets(0, 1, 11),
 		},
-		[]string{"instance_name_prefix", "platform", "size_class"})
+		[]string{},
+	}
 )
 
+// counterTemplate is a utility struct for generating and registering prometheus.CounterVec
+// resources from a set of options, and required labels to appear in the resulting vec.
+type counterTemplate struct {
```

Comment:
I'm wondering if there's a better place for this to live? It feels a little divorced from the rest of the logic in this file, which is already massive. Open to suggestions here.